footerTest.js

```diff
@@ -0,0 +1,70 @@
+const puppeteer = require('puppeteer');
+
+let config = {
+  launchOptions: {
+    headless: false,
+    ignoreHTTPSErrors: true,
+  },
+};
+
+let urls = [
+  'https://dashboard.realdevsquad.com/index.html',
+  'https://dashboard.realdevsquad.com/profile/index.html',
+  'https://dashboard.realdevsquad.com/goal/index.html',
+  'https://dashboard.realdevsquad.com/task/index.html',
+  'https://dashboard.realdevsquad.com/users/index.html',
+  'https://dashboard.realdevsquad.com/users/details/index.html',
+  'https://dashboard.realdevsquad.com/taskevents/index.html',
+  'https://dashboard.realdevsquad.com/featureflag/index.html',
+  'https://dashboard.realdevsquad.com/wallet/index.html',
+  'https://dashboard.realdevsquad.com/online-members/online-members.html',
+];
+
+const test_footer = async (url, index) => {
+  await puppeteer.launch(config.launchOptions).then(async (browser) => {
+    const page = await browser.newPage();
+    await page.goto(url, { waitUntil: ['load', 'networkidle2'] });
+    await page.content();
+    page.setDefaultNavigationTimeout(0);
+    const footer = await page.$('footer');
+    if (footer) {
+      // is the text same in footer?
+      const footer_text = await page.evaluate(() => {
+        const element = document.querySelector('.info-repo');
+        console.log(element);
+
+        return (
+          element &&
+          element.innerText ===
+            'The contents of this website are deployed from this open sourced repo'
+        );
+        // will return undefined if the element is not found
+      });
+      console.log(`✅ footer text:${footer_text}  index:${index}:${url}`);
+
+      if (footer_text) {
+        // links inside footer are working or not
+        await Promise.all([
+          page.click('.info-repo a', { delay: 2000 }),
+          page.waitForNavigation(),
+          setTimeout(() => {
+            console.log(`✅ link works for ${url}`);
+            browser.close();
+          }, 10000),
+        ]).catch((e) => console.log('disconnected '));
+      }
+    } else {
+      console.log(`❌no footer at ${url}`);
+      browser.disconnect();
+    }
+  });
+};
+
+async function runTest() {
+  for (let url of urls) {
```

Comment:
If I am reading this correctly, you are spinning up a new browser to test every url. Right?

While this works, there's a faster way using the "incognito" mode of browser. The browser is already loaded and ready to go, and we create a new isolated context per page, which loads faster (We should do some benchmarking around this!). See some examples of this [here](https://github.com/Real-Dev-Squad/website-tests/blob/ada05cc3d0a9f14923359d2c828e8b48988e4e3c/acceptance/new-user/www-loads.test.ts#L11).

Reply:
I did that but I was facing a problem closing the browser. will try again.